### DISCUSSION TITLE
Can't get Simplisafe Homebridge to stay connected.

### DISCUSSION THREAD

**Discussion by PoppyS33d** [homebridge-simplisafe3/homebridge-simplisafe3#371] [Q&A]
Hi there, I'm trying to get a fresh Mac Simplisafe Homebridge install to stay connected. On version 1.10.10 I couldn't even get authenticated, but rolling back to 1.10.08 I was able to, but it only seems to stay online for maybe an hour, if that, before all devices become unresponsive. I've since updated to the new 1.10.11 that came out in the next day and was able to authenticate again (seems like that whole process was reversed due to some internal Simplisafe changed), but - again, maybe half an hour later the Simplisafe alarm is "No Response" when I try to interact. 
<img width="1522" alt="Screenshot 2022-11-18 at 9 30 09 AM" src="https://user-images.githubusercontent.com/39997942/202766520-094ecfea-ef52-40d9-8cf0-b636a4695a03.png">

Not sure what else I can try, but any help is appreciated!

**Comment by shamoon** [comment]
The way you initially authenticate doesnt really have anything to do with staying authenticated. Have you changed any of the plugin settings, i.e. `sensorRefresh`?

**Comment by PoppyS33d** [comment]
I haven't touched any advanced settings of any sort. 
<img width="695" alt="Screenshot 2022-11-18 at 9 36 17 AM" src="https://user-images.githubusercontent.com/39997942/202767189-c19596e3-0b24-45ca-b55e-35eb0e01bac1.png">

Settings still says I'm authenticated, but clearly I'm not, right? I'm just at a loss. 

<img width="727" alt="Screenshot 2022-11-18 at 9 35 39 AM" src="https://user-images.githubusercontent.com/39997942/202767337-baa7956c-4295-493d-ab23-4950250383d3.png">

**Reply by shamoon** [reply]
Ok. One other thing is sometimes people end up running multiple instances of homebridge, thats worth checking for. Finally, I would enable `debug` on the plugin, turn it off for a day (yes) and try again tomorrow. Sometimes your IP gets temporarily banned by SS especially if you've been fiddling with logging in. Let me know how it goes after that (and if youre still seeing it the logs with `debug` might be helpful).

In general Im not aware of any current issues with this, seems things are humming along normally for most folks (again, separate from login issue as you noted)

**Reply by PoppyS33d** [reply]
Thanks for the reply. I did use to run Homebridge just from Command Line, but uninstalled it via the Command Line, - at least I believe so.   I'd also completely removed the .Homebridge folder before installing the GUI fresh, so that shouldn't be an issue as the configuration (config.json) was removed. 

Every error I seem to be getting since switching to the GUI is related to the 403, e.g. 

_[11/18/2022, 9:29:36 AM] [Home Alarm] SimpliSafe login failed with error: {
  message: 'Request failed with status code 403',_

I'll disable it for a day, and get further debug logs if I can tomorrow. Thank you for the help.

**Reply by PoppyS33d** [reply]
Good news: I wiped my entire Homebridge setup against and started fresh. It’s been 24 hours and still working. Woohoo! Thanks for the assistance!

**Reply by PoppyS33d** [reply]
And several days later it gave up with a 403 error. :( Authenticated again, but doing this every day or two isn't a great solution. Any other ideas? Thank you.

**Reply by shamoon** [reply]
If you enable `debug` the logs should show the plugin re-authenticating every ~55 minutes or so, and might perhaps show something when auth fails. But no, not really. Again, doesnt seem widespread so not sure if theres something specifically going on for you.

**Reply by PoppyS33d** [reply]
Thanks for the reply. Here are some of the errors with debug enabled:

11/20/2022, 9:38:38 PM] [Home Alarm] Recovery credentials refresh failed with error: {
  message: 'Request failed with status code 403',
  name: 'Error',
  description: undefined,
  number: undefined,
  fileName: undefined,
  lineNumber: undefined,
  columnNumber: undefined,
  stack: 'Error: Request failed with status code 403\n' +
    '    at createError (/usr/local/lib/node_modules/homebridge-simplisafe3/node_modules/axios/lib/core/createError.js:16:15)\n' +
    '    at settle (/usr/local/lib/node_modules/homebridge-simplisafe3/node_modules/axios/lib/core/settle.js:17:12)\n' +
    '    at IncomingMessage.handleStreamEnd (/usr/local/lib/node_modules/homebridge-simplisafe3/node_modules/axios/lib/adapters/http.js:322:11)\n' +
    '    at IncomingMessage.emit (node:events:539:35)\n' +
    '    at endReadableNT (node:internal/streams/readable:1345:12)\n' +
    '    at processTicksAndRejections (node:internal/process/task_queues:83:21)',
  config: {
    transitional: {
      silentJSONParsing: true,
      forcedJSONParsing: true,
      clarifyTimeoutError: false
    },
    adapter: [Function: httpAdapter],
    transformRequest: [ [Function: transformRequest] ],
    transformResponse: [ [Function: transformResponse] ],
    timeout: 0,
    xsrfCookieName: 'XSRF-TOKEN',
    xsrfHeaderName: 'X-XSRF-TOKEN',
    maxContentLength: -1,
    maxBodyLength: -1,
    validateStatus: [Function: validateStatus],
    headers: {
      Accept: 'application/json, text/plain, */*',
      'Content-Type': 'application/json',
      Host: 'auth.simplisafe.com',
      'Auth0-Client': '[redacted]',
      'User-Agent': 'axios/0.26.1',
      'Content-Length': 189
    },
    baseURL: 'https://auth.simplisafe.com/oauth',
    method: 'post',
    url: '/token',
    data: '{"grant_type":"refresh_token","client_id":"[redacted]","refresh_token":"[redacted]"}',
    'axios-retry': { retryCount: 0, lastRequestTime: 1669009118562 }
  },
  code: undefined,
  status: 403
}
[11/20/2022, 9:38:38 PM] [Home Alarm] Sensor refresh received an error from the SimpliSafe API: Error: 403: Forbidden
    at SimpliSafe3.request (/usr/local/lib/node_modules/homebridge-simplisafe3/simplisafe.js:211:17)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at SimpliSafe3.getSensors (/usr/local/lib/node_modules/homebridge-simplisafe3/simplisafe.js:363:32)
    at Timeout._onTimeout (/usr/local/lib/node_modules/homebridge-simplisafe3/simplisafe.js:702:25)

**Reply by shamoon** [reply]
Best thing would be to upload a dump of the logs, its not the 403 its everything leading up to it, again, maybe. The above suggests it already failed at least once before that snippet.

**Reply by PoppyS33d** [reply]
> Best thing would be to upload a dump of the logs, its not the 403 its everything leading up to it, again, maybe. The above suggests it already failed at least once before that snippet.

Where does one locate the rest of the log? That was just what I could see on the main page of the GUI.

**Reply by shamoon** [reply]
theres a download button. And its in your ~/.homebridge (or wherever) dir

**Reply by PoppyS33d** [reply]
Thank you, I got it. Is there anything private in the log I should be concerned about? I am seeing some odd messages in there... Let me grab a few...

**Reply by PoppyS33d** [reply]
This is an odd one:

[0;37m[11/20/2022, 9:54:46 PM] [0m[0;36m[Homebridge UI][0m [0;33mHomebridge Config UI X v4.50.2 is listening on :: port 8581[0m
[0;37m[11/20/2022, 9:54:46 PM] [0m[0;36m[Homebridge UI][0m [0;31mError: listen EADDRINUSE: address already in use :::8581[0m
[0;37m[11/20/2022, 9:54:46 PM] [0m[0;36m[Homebridge UI][0m [0;31mAnother process or service on this host is using port 8581.[0m
[0;37m[11/20/2022, 9:54:46 PM] [0m[0;36m[Homebridge UI][0m [0;31mPlease stop the other service or change the port you have assigned to homebridge-config-ui-x.[0m
[0;37m[11/20/2022, 9:54:46 PM] [0m[0;36m[Homebridge UI][0m [0;31mEnding process now.[0m

And...

11/20/2022, 9:57:06 PM] [Home Alarm] Error: 403: Forbidden
    at SimpliSafe3.request (/usr/local/lib/node_modules/homebridge-simplisafe3/simplisafe.js:211:17)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at SimpliSafe3.getUserId (/usr/local/lib/node_modules/homebridge-simplisafe3/simplisafe.js:252:16)
    at SimpliSafe3.getSubscriptions (/usr/local/lib/node_modules/homebridge-simplisafe3/simplisafe.js:261:18)
    at SimpliSafe3.getSubscription (/usr/local/lib/node_modules/homebridge-simplisafe3/simplisafe.js:289:18)
    at SimpliSafe3.getAlarmSystem (/usr/local/lib/node_modules/homebridge-simplisafe3/simplisafe.js:329:24)
    at SS3Alarm.getAlarmState (/usr/local/lib/node_modules/homebridge-simplisafe3/accessories/alarm.js:251:18)
    at SS3Alarm.refreshState (/usr/local/lib/node_modules/homebridge-simplisafe3/accessories/alarm.js:236:19)
[0;37m[11/20/2022, 9:57:07 PM] [0m[0;36m[Homebridge UI][0m [0;33mHomebridge Config UI X v4.50.2 is listening on :: port 8581[0m
[0;37m[11/20/2022, 9:57:07 PM] [0m[0;36m[Homebridge UI][0m [0;31mError: listen EADDRINUSE: address already in use :::8581[0m
[0;37m[11/20/2022, 9:57:07 PM] [0m[0;36m[Homebridge UI][0m [0;31mAnother process or service on this host is using port 8581.[0m
[0;37m[11/20/2022, 9:57:07 PM] [0m[0;36m[Homebridge UI][0m [0;31mPlease stop the other service or change the port you have assigned to homebridge-config-ui-x.[0m
[0;37m[11/20/2022, 9:57:07 PM] [0m[0;36m[Homebridge UI][0m [0;31mEnding process now.[0m
[11/20/2022, 9:57:15 PM] Loaded config.json with 0 accessories and 3 platforms.
[11/20/2022, 9:57:15 PM] Loaded 20 cached accessories from cachedAccessories.

**Reply by shamoon** [reply]
Either multiple instances running =) or something else bungled about your setup

**Reply by PoppyS33d** [reply]
> Either multiple instances running =) or something else bungled about your setup

I hear you, but this is a fresh install. I'm not sure how anything else can be running, and I'm not sure how to check for certainly either. :(

**Reply by shamoon** [reply]
im not sure how youre running it but id guess at some point you setup hb-service, see https://github.com/oznu/homebridge-config-ui-x/wiki/Homebridge-Service-Command

I'd start fresh. There's also a homebridge discord if youre having trouble with the basic setup https://discord.gg/kqNCe2D

**Reply by PoppyS33d** [reply]
> im not sure how youre running it but id guess at some point you setup hb-service, see https://github.com/oznu/homebridge-config-ui-x/wiki/Homebridge-Service-Command
> 
> I'd start fresh. There's also a homebridge discord if youre having trouble with the basic setup https://discord.gg/kqNCe2D

I did start fresh, though. I followed these instructions to a T:

https://github.com/homebridge/homebridge/wiki/Install-Homebridge-on-macOS

I also wiped everything, including the .homebridge folder, before following those instructions.

**Reply by PoppyS33d** [reply]
I also wanted to add that I'm using the TP-Link KASA Homebridge devices without any issue. Surely they'd not work either if it was a Homebridge problem? I nuked Homebridge for the 75th time, and can confirm it's not running. I don't know what else to do...

**Comment by jkossis** [comment]
@PoppyS33d @shamoon I am running into the same issue as well, getting a 403 roughly 3-4 days after I auth (happens consistently).

Any resolution to this discussion from a couple weeks back?

**Reply by shamoon** [reply]
Not really. These random failures are hard to diagnose, especially since I don’t experience them (in general I go weeks - months without having to re-auth). Assuming refresh isn’t set too low which will get you temporarily banned, you can check the logs (even enable `debug`) to make sure it’s preemptively re-authenticating ~1h or so or anything else that might shed light.

Sorry, the tough thing about a kinda flaky third-party APIs

**Reply by PoppyS33d** [reply]
I did a deep Mac clean of Homebridge and reinstalled. I’ve not had any issues since, but I can’t tell you what it was specifically that was causing problems.
